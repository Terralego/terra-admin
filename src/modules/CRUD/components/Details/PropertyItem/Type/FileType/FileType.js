import React from 'react';
import PropTypes from 'prop-types';
import { parseUrl } from 'query-string';
import NoValue from '../../NoValue';

const getFileName = src => {
  const { url } = parseUrl(src);
  return decodeURIComponent(url.split('/').pop());
};

const FileType = ({
  display_value: { url, thumbnail },
  schema: { title },
  type,
}) => {
  if (!url) {
    return <NoValue />;
  }

  const text = getFileName(url) || title;

  if (type === 'image') {
    return <img key={thumbnail} src={thumbnail} alt={text} />;
  }

  return (
    <a
      key={url}
      href={url}
      rel="download noopener noreferrer"
      target="_blank"
    >
      {text}
    </a>
  );
};

FileType.propTypes = {
  display_value: PropTypes.shape({
    thumbnail: PropTypes.string,
    url: PropTypes.string,
  }),
  schema: PropTypes.shape({
    title: PropTypes.string,
  }),
  type: PropTypes.string,
};

FileType.defaultProps = {
  display_value: {
    thumbnail: undefined,
    url: undefined,
  },
  schema: {
    title: '',
  },
  type: 'file',
};

export default FileType;
