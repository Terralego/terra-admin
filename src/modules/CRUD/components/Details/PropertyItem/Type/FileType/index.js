import FileType from './FileType';

export default FileType;
