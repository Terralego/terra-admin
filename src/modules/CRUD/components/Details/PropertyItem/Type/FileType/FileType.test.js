import React from 'react';
import renderer from 'react-test-renderer';
import FileType from './FileType';

jest.mock('../../NoValue', () => () => <div>No value</div>);

const props = {
  display_value: {
    url: 'http://file.pdf',
    thumbnail: 'http://file_thumb.jpg',
  },
  schema: { title: 'Name of the file' },
};

describe('should render correctly', () => {
  it('file', () => {
    const tree = renderer.create((
      <FileType {...props} />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('image', () => {
    const tree = renderer.create((
      <FileType {...props} type="image" />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('nothing', () => {
    const tree = renderer.create((
      <FileType display_value={{ url: undefined }} />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
