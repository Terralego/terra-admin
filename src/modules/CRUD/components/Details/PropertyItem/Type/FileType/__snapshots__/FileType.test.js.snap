// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly file 1`] = `
<a
  href="http://file.pdf"
  rel="download noopener noreferrer"
  target="_blank"
>
  file.pdf
</a>
`;

exports[`should render correctly image 1`] = `
<img
  alt="file.pdf"
  src="http://file_thumb.jpg"
/>
`;

exports[`should render correctly nothing 1`] = `
<div>
  No value
</div>
`;
